templates/injector-deployment.yaml

```diff
@@ -107,6 +107,8 @@ spec:
               value: "{{ .Values.injector.agentDefaults.memLimit }}"
             - name: AGENT_INJECT_DEFAULT_TEMPLATE
               value: "{{ .Values.injector.agentDefaults.template }}"
+            - name: AGENT_INJECT_TEMPLATE_CONFIG_EXIT_ON_RETRY_FAILURE
+              value: "{{ .Values.injector.agentDefaults.templateConfig.exitOnRetryFailure }}"
```

Comment:
If I add  `{{ ... | default true }}` the `injector/deployment: can set agent template_config.exit_on_retry_failure"` bats test fails, and I'm not sure why.

Reply:
Defaults should be only set in values.yaml since it's easier to keep track, so this is fine.

Reply:
I saw that `.Values.injector.revokeOnShutdown` has a default value so I wanted to give it a try. I'm still a bit curious on why `--set` on the test doesn't work if I add a default over there though.

Reply:
I suspect that's because revokeOnShutdown is a boolean, so if it's `false` that the same as it being unset, which is when the `| default true }` would be used.